scico/linop/radon_svmbir.py

```diff
@@ -230,15 +237,35 @@ def __init__(
         if prox_kwargs is None:
             prox_kwargs = dict()
 
+        default_prox_args = {"maxiter": 1000, "ctol": 0.001}
+        default_prox_args.update(prox_kwargs)
+
         svmbir_prox_args = dict()
-        if "maxiter" in prox_kwargs:
-            svmbir_prox_args["max_iterations"] = prox_kwargs["maxiter"]
-        if "ctol" in prox_kwargs:
-            svmbir_prox_args["stop_threshold"] = prox_kwargs["ctol"]
+        if "maxiter" in default_prox_args:
+            svmbir_prox_args["max_iterations"] = default_prox_args["maxiter"]
+        if "ctol" in default_prox_args:
+            svmbir_prox_args["stop_threshold"] = default_prox_args["ctol"]
         self.svmbir_prox_args = svmbir_prox_args
 
         self.positivity = positivity
 
+        if W is None:
+            self.W = Identity(self.y.shape)
+        elif isinstance(W, Diagonal):
+            if snp.all(W.diagonal >= 0):
+                self.W = W
+            else:
+                raise Exception(f"The weights, W.diagonal, must be non-negative.")
+        else:
```

Comment:
Perhaps just "The weights, W, must be ..." since W.diagonal is internal and not part of the interface?

Reply:
Fixed.